### PR TITLE
Fixes issue #315: 'this' is saved before use

### PR DESCRIPTION
**Pull request by solomax**
Monkey-patching due to original plugin seems to be abandoned

### CODE REVIEW

**Review by martin-g — APPROVED** [review]

File: wicket-jquery-ui-plugins/src/main/resources/com/googlecode/wicket/jquery/ui/plugins/wysiwyg/resource/js/bootstrap-wysiwyg.min.js [hunk 1]
```diff
@@ -6,5 +6,4 @@
...
...his.value="";self.restoreSelection();if(newValue){editor.focus();self.execCommand($(this).data(options.commandRole),newValue,editor,options,toolbarBtnSelector)}self.saveSelection()}).on("focus",function(){var input=$(this);if(!input.data(options.selectionMarker)){self.markSelection(input,options.selectionColor,options);input.focus()}}).on("blur",function(){var input=$(this);if(input.data(options.selectionMarker)){self.markSelection(input,false,options)}});toolbar.find("input[type=file][data-"+options.commandRole+"]").change(function(){self.restoreSelection();if(this.type==="file"&&this.files&&this.files.length>0){self.insertFiles(this.files,options,editor,toolbarBtnSelector)}self.saveSelection();this.value=""})};Wysiwyg.prototype.initFileDrops=function(editor,options,toolbarBtnSelector){var self=this;editor.on("dragenter dragover",false).on("drop",function(e){var dataTransfer=e.originalEvent.dataTransfer;e.stopPropagation();e.preventDefault();if(dataTransfer&&dataTransfer.files&&dataTransfer.files.length>0){self.insertFiles(dataTransfer.files,options,editor,toolbarBtnSelector)}})};$.fn.wysiwyg=function(userOptions){var wysiwyg=new Wysiwyg(this,userOptions)}})(window,window.jQuery);
```
Comment (martin-g) [inline wicket-jquery-ui-plugins/src/main/resources/com/googlecode/wicket/jquery/ui/plugins/wysiwyg/resource/js/bootstrap-wysiwyg.min.js:9]: Did you minimize it manually ? :-) (s/\n//g ?!)

Comment (solomax) [reply wicket-jquery-ui-plugins/src/main/resources/com/googlecode/wicket/jquery/ui/plugins/wysiwyg/resource/js/bootstrap-wysiwyg.min.js:9]: @martin-g I run `uglify` from command line and add header manually :(
Will create pr to original repo ...

Comment (solomax) [reply wicket-jquery-ui-plugins/src/main/resources/com/googlecode/wicket/jquery/ui/plugins/wysiwyg/resource/js/bootstrap-wysiwyg.min.js:9]: PR created https://github.com/steveathon/bootstrap-wysiwyg/pull/153

**Comment by sebfz1** [comment]
Thanks Maxim and Martin! :)
I'll do the minimized version, merge and release asap...

**Comment by solomax** [comment]
Thanks @sebfz1 ! :)
(was sure command line `uglify` should work as expected :(( )

**Comment by sebfz1** [comment]
`uglifyjs bootstrap-wysiwyg.js -cm -o bootstrap-wysiwyg.min.js`
I believe the "mangle" option was missing

**Comment by solomax** [comment]
Always thought it is something from "Harry Potter" ;)
Thanks again :)

**Comment by sebfz1** [comment]
> Always thought it is something from "Harry Potter" ;)

Haha, was my impression too! :)
Merged as 3a9b264ad66ab5b1dfaea814e41e784dfa43ca8e

**Comment by solomax** [comment]
Hello @sebfz1 
When do you plan to deploy maven artifacts? :)

**Comment by solomax** [comment]
BTW, maybe it worth to cherry-pick this fix to 8.x?

**Comment by sebfz1** [comment]
Already cherry picked :)
I will release M5.1 today...

>

**Pull request closed without merging**